chain/substreams/src/trigger.rs

```diff
@@ -85,3 +100,133 @@ impl blockchain::TriggersAdapter<Chain> for TriggersAdapter {
+                    return Err(MappingError::Unknown(anyhow!("Detected UNSET entity operation, either a server error or there's a new type of operation and we're running an outdated protobuf")));
+                }
+                Operation::Create | Operation::Update => {
+                    let entity_type: &str = &entity_change.entity;
+                    let entity_id: String = String::from_utf8(entity_change.id.clone())
+                        .map_err(|e| MappingError::Unknown(anyhow::Error::from(e)))?;
+                    let key = EntityKey {
+                        subgraph_id: self.locator.hash.clone(),
+                        entity_type: EntityType::new(entity_type.into()),
+                        entity_id: entity_id.clone(),
+                    };
+
+                    let mut data: HashMap<String, Value> = HashMap::from_iter(vec![]);
+                    for field in entity_change.fields.iter() {
+                        let value: Value = match (field.value_type(), field.new_value_null) {
+                            (Type::Unset, _) => {
+                                return Err(MappingError::Unknown(anyhow!(
+                                    "Invalid field type, the protobuf probably needs updating"
+                                )))
+                            }
+                            (Type::Bigdecimal, false) => {
+                                match BigDecimal::parse_bytes(field.new_value.as_ref()) {
+                                    Some(bd) => Value::BigDecimal(bd),
+                                    None => {
+                                        return Err(MappingError::Unknown(anyhow!(
+                                            "Unable to parse BigDecimal for entity {}",
+                                            entity_change.entity
+                                        )))
+                                    }
+                                }
+                            }
+                            (Type::Bigint, false) => Value::BigInt(BigInt::from_signed_bytes_be(
+                                field.new_value.as_ref(),
+                            )),
+                            (Type::Int, false) => {
+                                let mut bytes: [u8; 4] = [0; 4];
+                                bytes.copy_from_slice(field.new_value.as_ref());
+                                Value::Int(i32::from_be_bytes(bytes))
+                            }
+                            (Type::Bytes, false) => {
+                                Value::Bytes(Bytes::from(field.new_value.as_ref()))
+                            }
+                            (Type::String, false) => Value::String(
+                                String::from_utf8(field.new_value.clone())
+                                    .map_err(|e| MappingError::Unknown(anyhow::Error::from(e)))?,
+                            ),
+                            _ => {
```

Comment:
Aren't we missing the handling of `(_, true)` case which means the field is not set? Or maybe we are already using field presence via `optional` keyword and this is waiting to be removed from the proto?

Reply:
I see we talk about `null` value in the default case below, are we not dealing with `null` value to "unset" a field?